gnome-initial-setup/gis-driver.c

```diff
@@ -288,6 +294,208 @@ gis_driver_get_mode (GisDriver *driver)
+
+  g_object_unref (power_settings);
+}
+
+#define GS_SCHEMA "org.gnome.software"
+#define GS_ALLOW_UPDATES "allow-updates"
+
+void
+configure_demo_mode_disallow_app_center_updates ()
+{
+  GSettings *gnome_software_settings = g_settings_new (GS_SCHEMA);
+
+  g_settings_set_boolean (gnome_software_settings, GS_ALLOW_UPDATES, FALSE);
+
+  g_object_unref (gnome_software_settings);
+}
+
+#define GNOME_SHELL_SCHEMA "org.gnome.shell"
+#define GNOME_SHELL_FAVORITE_APPS "favorite-apps"
+
+#define GOOGLE_CHROME_DESKTOP_ID "google-chrome.desktop"
+#define CHROMIUM_BROWSER_DESKTOP_ID "chromium-browser.desktop"
+
+void
+replace_string_in_strv (GStrv       strv,
+                        const gchar *needle,
+                        const gchar *replacement)
+{
+  guint length = g_strv_length (strv);
+  guint i = 0;
+
+  for (; i < length; ++i)
+    {
+      if (g_strcmp0 (strv[i], needle) == 0)
+        {
+          g_free (strv[i]);
+          strv[i] = g_strdup (replacement);
+        }
+    }
+}
+
+void
+replace_google_chrome_in_favorite_apps (GSettings *gnome_shell_settings)
+{
+  GStrv favorite_apps = g_settings_get_strv (gnome_shell_settings,
+                                             GNOME_SHELL_FAVORITE_APPS);
+  replace_string_in_strv (favorite_apps,
+                          GOOGLE_CHROME_DESKTOP_ID,
+                          CHROMIUM_BROWSER_DESKTOP_ID);
+  g_settings_set_strv (gnome_shell_settings,
+                       GNOME_SHELL_FAVORITE_APPS,
+                       (const gchar * const *) favorite_apps);
+
+  g_strfreev (favorite_apps);
+}
+
+void
+configure_demo_mode_replace_chrome_with_chromium ()
+{
+  GSettings *gnome_shell_settings = g_settings_new (GNOME_SHELL_SCHEMA);
+  replace_google_chrome_in_favorite_apps (gnome_shell_settings);
+  g_object_unref (gnome_shell_settings);
+}
+
+gboolean
+setup_demo_config (GisDriver *driver, GError **error)
+{
+  gchar *stamp_file = g_build_filename (g_get_user_config_dir (),
+                                        "eos-demo-mode",
+                                        NULL);
+
+  if (!g_file_set_contents (stamp_file, "1", sizeof(char), error))
+    {
+      g_free (stamp_file);
+      return FALSE;
+    }
+
+  configure_demo_mode_power_settings ();
+  configure_demo_mode_disallow_app_center_updates ();
+  configure_demo_mode_replace_chrome_with_chromium ();
+
+  g_free (stamp_file);
+  return TRUE;
+}
+
+void
+gis_driver_enter_demo_mode (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+
+  if (priv->is_in_demo_mode)
+    return;
+
+  /* Enter test mode.
+   *
+   * After this point, all data on the system will become ephemeral,
+   * including /home, which means any changes will be lost on reboot.
+   */
+  GError *error = NULL;
+  if (!gis_pkexec (LIBEXECDIR "/eos-demo-mode", "demo-guest", NULL, &error))
+    {
+      handle_demo_mode_error (error);
+      return;
+    }
+
+  if (!create_demo_user (driver))
```

Comment:
Yeah I was thinking that for consistency's sake we should do this. Must have forgotten to.